scripts/check_branch.py

```diff
@@ -47,7 +44,7 @@ def check_branch(branch: Optional[str], stage: str) -> None:
             )
     else:
         assert branch is not None
-        if stage != expected_stage:
+        if stage != expected_stage and stage in stage_by_branch.values():
```

Comment:
I noticed that `config.is_main_deployment` is basically doing the same thing as the check you added.

Consider (this is completely optional and only at your discretion) consolidating the two.

Reply:
I think it makes sense to keep it as is since there's no guarantee that `stage` is `self.deployment_stage` (which `config.is_main_deployment`) checks. But I didn't know about `config.is_main_deployment` and I'll remember it for next time!

Reply:
Agree with @jessebrennan. There is room for consolidation.

```diff
Index: src/azul/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/azul/__init__.py	(revision b8fff54d12d4b6e18cf659996f6674bd9e1e0058)
+++ src/azul/__init__.py	(date 1593133817469)
@@ -465,9 +465,10 @@
         'prod': 'prod'
     }
 
-    @property
-    def is_main_deployment(self):
-        return self.deployment_stage in self.main_deployments_by_branch.values()
+    def is_main_deployment(self, stage=None):
+        if stage is None:
+            stage = self.deployment_stage
+        return stage in self.main_deployments_by_branch.values()
 
     @property
     def _git_status(self) -> Mapping[str, str]:
Index: terraform/s3.tf.json.template.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- terraform/s3.tf.json.template.py	(revision b8fff54d12d4b6e18cf659996f6674bd9e1e0058)
+++ terraform/s3.tf.json.template.py	(date 1593131564372)
@@ -20,7 +20,7 @@
                 },
                 "url_bucket": {
                     "bucket": config.url_redirect_full_domain_name,
-                    "force_destroy": not config.is_main_deployment,
+                    "force_destroy": not config.is_main_deployment(),
                     "acl": "public-read",
                     "website": {
                         # index_document is required; pointing to a non-existent file to return a 404
Index: test/integration_test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test/integration_test.py	(revision b8fff54d12d4b6e18cf659996f6674bd9e1e0058)
+++ test/integration_test.py	(date 1593131580944)
@@ -441,7 +441,7 @@
         notifications = [invalid_notification]
         self.assertRaises(AzulClientNotificationError, self.azul_client._index, notifications)
 
-    @unittest.skipIf(config.is_main_deployment, 'Test would pollute portal DB')
+    @unittest.skipIf(config.is_main_deployment(), 'Test would pollute portal DB')
     def test_concurrent_portal_db_crud(self):
         """
         Use multithreading to simulate multiple users simultaneously modifying
```